build.gradle

```diff
@@ -1,98 +1,76 @@
-buildscript {
-  repositories {
-    jcenter()
-  }
-}
-
 plugins {
     id 'java'
     id 'application'
     id 'base'
     id 'nebula.lint' version '9.3.4'
 }
-
-allprojects {
-  apply plugin: 'java'
-  apply plugin: 'base'
-  apply plugin: 'nebula.lint'
-  
-  repositories {
-    jcenter()
-  }
-
-  group = 'org.apereo.openequella.adminconsole'
-  version = artifactVersion
-
-  gradleLint.rules = ['all-dependency', 'dependency-parentheses', 'duplicate-dependency-class']
-
-  dependencies {
-    implementation 'org.slf4j:slf4j-api:1.7.26'
-    runtime 'org.slf4j:slf4j-simple:1.6.4'
-  }
-}
-
+/*
 subprojects {
   task sourcesJar(type: Jar) {
         from sourceSets.main.java
         archiveClassifier = 'sources'
     }
-}
+} */
 
-project(':common') {
-  dependencies {
-      implementation 'com.fasterxml.jackson.core:jackson-databind:2.9.8'
-      implementation 'org.jasypt:jasypt:1.9.2'
-  }
+repositories {
+  jcenter()
 }
 
-project(':configurator') {
-
-  mainClassName = 'org.apereo.openequella.adminconsole.configurator.Configurator'
+group = 'org.apereo.openequella.adminconsole'
+version = artifactVersion
+mainClassName = 'org.apereo.openequella.adminconsole.launcher.ClientLauncher'
 
-  apply plugin: "application"
-  
-  dependencies {
-    compile project(':common')
+apply plugin: 'java'
+apply plugin: 'base'
+apply plugin: 'nebula.lint'
+apply plugin: "application"
 
-    testImplementation "junit:junit:$junitVersion"
-  }
+// Gradlelint is broken: https://github.com/nebula-plugins/gradle-lint-plugin/issues/203
+gradleLint.rules = [/*'all-dependency', */'dependency-parentheses', 'duplicate-dependency-class']
 
-  // FIXME: gradle complains if this isn't listed in both positions...
-  mainClassName = 'org.apereo.openequella.adminconsole.configurator.Configurator'
+dependencies {
+  implementation 'org.slf4j:slf4j-api:1.7.26'
+  implementation 'com.fasterxml.jackson.core:jackson-databind:2.9.8' 
+  implementation 'org.jasypt:jasypt:1.9.2'
+  runtimeOnly 'org.slf4j:slf4j-simple:1.6.4'
+  testImplementation "junit:junit:$junitVersion"
+}
 
-  jar {
-    manifest {
+jar {
+  //dependsOn 'clean', 'build'
+  archiveFileName = 'admin.jar'
+  manifest {
       attributes('Implementation-Title': project.name,
               'Implementation-Version': project.version,
               'Main-Class': mainClassName)
-    }
   }
 }
 
-project(':launcher') {
-
-  mainClassName = 'org.apereo.openequella.adminconsole.launcher.ClientLauncher'
+task cleanDist() {
+  dependsOn 'clean'
+  doLast {
+    delete 'dist'
+  }
+}
 
-  apply plugin: "application"
+task copyDistArtifacts(type: Copy) {
+  dependsOn 'copyScripts'
 
-  group = 'org.apereo.openequella.adminconsole'
+  from(file('build/libs/admin.jar'))
+  from configurations.default
+  into('dist/lib')
```

Comment:
I may be wrong, but I wonder if this copy work should be put into a doLast/doFirst block because in the configuration stage admin.jar is not ready yet.

Reply:
Apparent when you declare a task like this it doesn't run any of it at configuration stage.  https://stackoverflow.com/a/30637190/877610

Reply:
OK. 
Just one more question here :) 
Can  we use 'from jar' instead of  'from(file('build/libs/admin.jar'))' ?  
Thank you!